### PR TITLE
Reworked codechef scraper

### PR DESCRIPTION
**Pull request by LaRuim**
Added a waiter to codechef scraper to replace while loop; converted the scraper into function, with a view to call it from an executor python file.

Added utility file for minor functions such as logging, rankfile naming, etc;

### CODE REVIEW

**Review by souravtecken — COMMENTED** [review]
@LaRuim, What exactly is the workflow here? The entire utitlities file seems to be specific for the codechef scraper. Also, since this is a step towards unifying the scrapers, can't we handle the file writing in the common script, this method would involve having file handling operations in all scrapers, we could instead have the scraper return the data, right? 

Also, use pythonic naming conventions? xD, I'm a fan of camel case too, but the rest of the repo follows the pythonic conventions, it's better to stick with it.

**Comment by LaRuim** [comment]
@souravtecken utilities.py was a file I created for convenience when working with selenium before, and I added functions I thought might be pertinent to the scraper there as well. 
In any case, I've split it based on functionality (into selenium_utils.py and log.py), and moved the codechef-specific functions to the scraper itself. As to why there is still a selenium_utils file; In my opinion, adding it to the scraper sorta bloats the code unnecessarily, but the functionality is still useful. 

The scraper now returns the data as a list, and the file handling will be done in the common script, which I will commit once you approve this

File: utils/selenium_utils.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+from selenium import webdriver
+from selenium.webdriver.chrome.options import Options
+from selenium.webdriver.support.wait import WebDriverWait
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support import expected_conditions as EC
+from time import strftime, sleep
+
+month = ['jan', 'feb', 'march', 'april', 'may', 'june', 'july', 'aug', 'sept', 'oct', 'nov', 'dec']
```
Comment (souravtecken) [inline utils/selenium_utils.py]: Place this in the unified scrapers file

File: scrapers/codechef.py [hunk 2]
```diff
@@ -1,57 +1,117 @@
...
+        easy_points=len(divisions['B'].problems-divisions['A'].problems)*easy_points # Points to add to div-A participants
+
+        for i in range(len(divisions['A'].scraped_scoreboard)): # Add easy points to all div-A participants
+            divisions['A'].scraped_scoreboard[i]=divisions['A'].scraped_scoreboard[i][0],divisions['A'].scraped_scoreboard[i][1]+easy_points
+            
+        final_scoreboard.extend(divisions['A'].scraped_scoreboard)
+        final_scoreboard.extend(divisions['B'].scraped_scoreboard) # Consolidate scoreboard
+        final_scoreboard = sorted(final_scoreboard,key=lambda x:x[1],reverse=True) # Sort list in desc. order based on points. (username, points)
+
+        if contest_code[0:5] == 'LTIME':
```
Comment (souravtecken) [inline scrapers/codechef.py]: This should probably be moved to the unified scrapers file too, let the only job of the scraper file be to fetch the rankings, offload everything else to the main file. The task of writing has already been offloaded to that, so I think this should too.

File: scrapers/codechef.py [hunk 3]
```diff
@@ -1,57 +1,117 @@
...
+                        shared_rank.append(user[0])
+            return contest_ranks_file, rank_list
+        #info(f'Leaderboard written to {contest_ranks_file}')
         print()
-        lowest_so_far = rank
-    print(handle, end=' ')
+    print()
+    driver.close()
+
+def find_month_year(offSet, ID, start):
```
Comment (souravtecken) [inline scrapers/codechef.py]: Some documentation required here, and move this to the main file too

**Review by souravtecken — CHANGES_REQUESTED** [review]

File: scrapers/codechef.py (hunk 3 above)
Comment (LaRuim) [reply scrapers/codechef.py]: this is used only for the codechef scraper though

File: scrapers/codechef.py (hunk 2 above)
Comment (LaRuim) [reply scrapers/codechef.py]: this also is a codechef-specific thing; none of the other platforms require intricate renaming..

Comment (LaRuim) [reply scrapers/codechef.py]: okay; will do anyway

**Comment by LaRuim** [comment]
@souravtecken excusez moi

**Review by souravtecken — APPROVED** [review]

**Pull request merged**